relayer/application_relayer.go

```diff
@@ -205,7 +206,9 @@ func (r *ApplicationRelayer) ProcessMessage(handler messages.MessageHandler) (co
 
 	// sourceWarpSignatureClient is nil iff the source blockchain is configured to fetch signatures via AppRequest
 	if r.sourceWarpSignatureClient == nil {
+		logContext := handler.GetLogContext(r.destinationClient)
```

Comment:
I think a better strategy might be that when we instantiate the `MessageHandler` instance, we call `r.logger.With(stuff we want to log)` to populate the log field. I think doing this whenever we create a child object that we want to link back to the parent would help

Reply:
Yeah but for this case to pass it along to the aggregator we would still need to reach into the `MessageHandler` to get this logger instance. Using a more specific structs  logger instance (in this case `MessageHandler`) directly seems more limiting since it would override any potential existing loggers fields. 

Returning just `zap.Field`s seems cleaner to me but point taken that we might as well be adding this context to `MessageHandler` internal logs.

Reply:
Actually just realized that in addressing feedback I've completely lost the point of adding the `GetLongContext` interface which was to log the `teleporterMessageID` as well which requires the `destinationClient` to compute. 

 I should either add that back or remove the interface changes entirely and just leave the changes on the aggregator side.